src/App.vue

```diff
@@ -35,8 +35,31 @@ export default {
     return {
       showKeycloakTools: false
     };
+  },
+  mounted() {
+    const missingRoles = getMissingRoles(
+        ["view-events", "manage-users", "view-users", "query-clients", "query-groups", "query-users"],
+        this.$keycloak.tokenParsed.resource_access['realm-management'].roles
+    );
+    if (missingRoles.length > 0) {
+      this.$store.commit("alert/setAlert", {
+        message: `You are missing roles required to use this application.
+          Missing roles: ${missingRoles}.`,
+        type: "error"
+      });
+    }
   }
 };
+
+function getMissingRoles(rolesRequired, actualRoles) {
+  const missingRoles = [];
+  rolesRequired.forEach(requiredRole => {
+    if (!actualRoles.includes(requiredRole)) {
+      missingRoles.push(requiredRole);
+    }
+  });
+  return missingRoles;
+}
```

Comment:
How to structure Javascript code is new territory for me. I created this function, I think it's "module scoped". I actually wanted to move the code in "mounted" into its own method too, but I don't know where to put it. If you make it a function like this one, you'd need to pass in references to Keycloak and Vuex. If you put it in the Vue Component's "methods" property, I think it will be called reactively by the framework during rendering.

Reply:
discussed on teams. Implementation makes sense to me